e2e-pw/src/oss/poms/field-visibility/field-visibility.ts

```diff
@@ -24,13 +58,120 @@ export class FieldVisibilityPom {
     return this.sidebarLocator.getByTestId("field-visibility-icon");
   }
 
+  get fieldVisibilityToggleTooltip() {
+    return this.page.getByText("Change field visibility");
+  }
+
+  get clearBtn() {
+    return this.sidebarLocator.getByTestId("field-visibility-btn-clear");
+  }
+
+  get applyBtn() {
+    return this.containerLocator.getByTestId("field-visibility-btn-apply");
+  }
+
+  get resetBtn() {
+    return this.containerLocator.getByTestId("field-visibility-btn-reset");
+  }
+
+  get filterRuleContainer() {
+    return this.containerLocator.getByTestId("filter-rule-container");
+  }
+
+  get filterRuleInput() {
+    return this.containerLocator.getByTestId(
+      "filter-visibility-filter-rule-input"
+    );
+  }
+
+  getFieldCheckbox(name: string) {
+    return this.page
+      .getByTestId(`schema-selection-${name}`)
+      .getByRole("checkbox");
+  }
+
+  getFieldVisibilityControl(label: string) {
+    return this.containerLocator.getByTestId(
+      `field-visibility-controls-${label}`
+    );
+  }
+
+  getControl(name: "select-all" | "show-metadata" | "show-nested-fields") {
+    return this.getFieldVisibilityControl(name);
+  }
+
+  getFieldInfoContainer(path: string) {
+    return this.containerLocator.getByTestId(
+      `schema-selection-info-container-${path}`
+    );
+  }
+
+  getTab(tabName: TabType) {
+    return this.containerLocator.getByTitle(tabName);
+  }
+
+  async getSelectionFields(
+    status: "checked" | "unchecked" | "all" = "checked",
+    mode:
+      | "parents-only"
+      | "nested-only"
+      | "all"
+      | "customFields" = "parents-only"
+  ) {
+    let paths: string[] = [];
+    switch (mode) {
+      case "parents-only":
+        paths = allParentPaths;
+        break;
+      case "nested-only":
+        paths = allSubpaths;
+        break;
+      case "all":
+        paths = allPaths;
+        break;
+      case "customFields":
+      default:
+        break;
+    }
+
+    const checked = status === "checked";
+    const fields: string[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      const cc = this.getFieldCheckbox(paths[i]);
+      if (status === "all") {
+        fields.push(cc);
```

Comment:
`fields` is a string array, but a `Locator` is appended here
```suggestion
        fields.push(paths[i]);
```